### PR TITLE
T279551 Plugin linting and CI

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator Ticket: https://phabricator.wikimedia.org/T279551

 - [x] Linter for JS
 - [x] Linter for php
 - [x] Connect `phpcs` to `wpcs`
 - [x] Fix php style

### CODE REVIEW

**Comment by hueitan** [comment]
Note:

1. The variable `$version` is used as this is the requirement from linter
2. I exclude the linter for `WordPress-Docs` because at this point i think doesn't make sense to write full doc
3. I'm new to php composer, so if you see anything doesn't make sense or break the common sense or practice, I would like to hear and do the feedback.

File: assets/js/init.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-wikipediaPreview.init({
-	root: document.querySelector('.entry-content'),
-	detectLinks: true
-});
+// eslint-disable-next-line no-undef
+wikipediaPreview.init( {
```
Comment (stephanebisson) [inline assets/js/init.js:2]: `wikipediaPreview` should be defined as a global in the eslint config

**Review by stephanebisson — CHANGES_REQUESTED** [review]
Looks good. One small thing to consider.

File: assets/js/init.js (hunk 1 above)
Comment (hueitan) [reply assets/js/init.js:1]: make sense and done.

**Pull request merged**